### ISSUE TITLE
[Accessibility] Narrator cannot announce some cursor items with specific names 

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#9464]
### .NET version

.Net 8.0 SDK: 8.0.100-preview.7.23357.8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, it is not a regression issue

### Issue description

**Narrator** cannot announce some cursor items with specific names 
![image](https://github.com/dotnet/winforms/assets/26474449/15bc709d-9ff7-4aee-aa57-ca0b7f18e678)
![CursorIssue](https://github.com/dotnet/winforms/assets/26474449/1476b86b-cd3e-4fe8-ad05-18a7287c0ce5)
**Inspect:**
![image](https://github.com/dotnet/winforms/assets/26474449/6138ee23-67b6-49f9-a140-e272180dfc3f)
**AccessibilityInsight:**
![image](https://github.com/dotnet/winforms/assets/26474449/9e79335e-04a0-415f-ae77-439916e3e590)

### Steps to reproduce

1. Create a Winforms .Net Core application with PropertyGrid control
2. Go to propertyGrid's properties window, find SelectedObject property, and set its value to Form1
3. Build and run application
4. Use Accessibility tools to test Cursor property in PropertyGrid

or use following sample:
[CursorSample.zip](https://github.com/dotnet/winforms/files/12012104/CursorSample.zip)

**Comment by elachlan** [comment]
@Olina-Zhang can you reproduce this in WinFormsTest PropertyGrid?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang can you reproduce this in WinFormsTest PropertyGrid?

No, it doesn't reproduce that app in winforms repo. And reproduce in new created application.

**Comment by elachlan** [comment]
The editor for cursor looks different in `WinFormsTest` `PropertyGrid`. 

I don't think any of the System.Drawing.Design` editors are working:
![image](https://github.com/dotnet/winforms/assets/2433737/1ebededc-d73b-4691-9c6c-90ff9d88539e)

**Comment by elachlan** [comment]
The cursors that are not showing use a different constructor which set `_ownHandle = false;`
https://github.com/dotnet/winforms/blob/8a825edb385efa8f4c0696af7cf9bc1094218ff5/src/System.Windows.Forms/src/System/Windows/Forms/Cursors.cs#L68-L70

Then the `ToString` method on `Cursor` calls the `CursorConverter` to get the value only if `_ownHandle=false`.
https://github.com/dotnet/winforms/blob/8a825edb385efa8f4c0696af7cf9bc1094218ff5/src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs#L476-L486

Changing the code to this fixed it:
```csharp
public override string ToString() => $"[Cursor: {TypeDescriptor.GetConverter(typeof(Cursor)).ConvertToString(this)}]";
```

**Comment by Olina-Zhang** [comment]
> The editor for cursor looks different in `WinFormsTest` `PropertyGrid`.

In WinformsTest propertyGrid, there is no icons for any of cursor, also repro in other property: Color
Cursor:
![image](https://github.com/dotnet/winforms/assets/26474449/d4069a7b-503e-4980-a158-7a9bc3d78b9a)
Color:
![image](https://github.com/dotnet/winforms/assets/26474449/ac41e581-e224-4d0d-b341-64825314a670)

**Comment by elachlan** [comment]
@Olina-Zhang I think all `System.Drawing.Design` editors aren't working in the PropertyGridTest and I am unsure why.

They would be:
- BitmapEditor
- ColorEditor
- ContentAlignmentEditor
- CursorEditor
- FontEditor
- FontNameEditor
- IconEditor
- ImageEditor
- MetafileEditor

**Comment by elachlan** [comment]
@Olina-Zhang `WinFormsTest` should work now.

**Comment by Olina-Zhang** [comment]
@elachlan Yes, they work now.

**Comment by elachlan** [comment]
@merriemcgaw the PR did not resolve this issue? Was this because it was merged to .NET 9? Should we just resolve this?

**Comment by merriemcgaw** [comment]
I bet the fact that we merged into the .NET 9 feature branch rather than main is why the tooling didn't automatically close this. I think what makes sense is when we go to merge the branch into main @Olina-Zhang and her amazing team double check the bugs and close the issues that went into it. I'm actually curious if the tools will pick up that the commits are in main and auto-close it anyway.

**Comment by Olina-Zhang** [comment]
Verified this issue in feature/9.0 branch, it contains the fixing now. Closing this issue manually.
Inspect:
![image](https://github.com/dotnet/winforms/assets/26474449/2148f934-37ef-4db0-bcb9-1f6a3e12827e)
Narrator:
![image](https://github.com/dotnet/winforms/assets/26474449/b6e8c63e-e884-4c93-8143-e99648f32145)
